---
title: "R Notebook for processing MRC clade metagenome abundance data from NRS time series"
output: html_notebook
---

## Authors: Martin Ostrowski, James O'Brien
## Date: 20190404
## email: martin.ostrowski@uts.edu.au


```{r}
library(tidyverse)
library(tidyquant)
```


Import unique filtered blast table

Import the gene catalog (add taxonomy and clade annotations, and gene annotations )

Import the taxonomic classification

Import the metadata
```{r}


mai<-read_tsv("~/Dropbox/AMI/MRC.bins/header.txt")
phb<-read_tsv("~/Dropbox/AMI/MRC.bins/headers.txt")
nsi<-read_tsv("~/Dropbox/AMI/MRC.bins/headers2.nsi.txt")
mrc<-read_tsv("~/Dropbox/AMI/MRC.bins/mrc.gene.catalog.txt")
mrc.clade<-read_tsv("~/Dropbox/AMI/MRC.bins/clades.v2.gtdb.tsv")
ami.dates<-read_tsv("~/Dropbox/AMI/MRC.bins/ami_nrs_dates.tsv")

```
first, get an overview of the available metagenomes for the NRS time series (as of March 2019)

```{r}
#convert the numeric months to a factor and set the levels
ami.dates$month.abb <-factor(month.abb[as.numeric(ami.dates$month)], levels=c("Jan", "Feb","Mar", "Apr","May",  "Jun", "Jul","Aug","Sep","Oct","Nov", "Dec" ))

#the site_codes can be converted to a factor with levels that will ensure things appear in the right order, from North to South

ami.dates$site_code<-factor(ami.dates$site_code, levels=c("DAR", "YON", "NSI", "PHB", "ROT", "KAI", "MAI"))

# a facet grid plot is great for getting things in order [note the reference to the edited theme theme_mo()]

ggplot(ami.dates, aes(x=month.abb, y=depth, col=site_code))+ geom_point()+facet_grid(site_code ~ year) + theme_mo() +theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

Prepare a plot of the relative abundance of defined MRC clades at MAI, PHB and NSI. 

First, join the blast table and annotation, split based on sample, split based on code, year, month, (depth)

As ingested, the data could be used for a genome recruitment plot of reads against the dominant genome scaffolds

These steps will likely need to be calculated on a hpc node with a lot of RAM
```{r}

mai<-rbind(mai, phb, nsi)

mai.mrc<- mai %>% left_join(mrc, 'subject')

mai.mrc <- mai.mrc %>% separate(query, c("sample", "read"), sep=";", remove=T)

mai.mrc <- mai.mrc %>% separate(sample, c("site_code", "Year", "Month", "Day", "depth"), sep=c(3,7,9,11), remove=F)

mai.mrc.sub1 <- mai.mrc %>% right_join(mrc.clade, c('strain'='strain')) # a right join leaves out the reads that are not assigned to clades that we have defined


mai.mrc.sub1$site_code<-factor(mai.mrc.sub1$site_code, levels=c("DAR", "YON", "NSI", "PHB", "ROT", "KAI", "MAI"))
table(mai.mrc.sub$clade)

#tidy up if memory is low. Find a way to do this more efficiently
#rm(mai.mrc)
#rm(phb)
#rm(nsi)
```


Visualise using a barplot

```{r, fig.width=10, fig.height=8}

p<-ggplot(mai.mrc.sub1 %>% filter (site_code %in% c("MAI", "PHB","NSI")), aes(x=Month, fill=clade)) + 
  geom_bar(stat="count", position="fill") + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) + scale_fill_manual(values=rev(mycol60)) + theme(legend.position="bottom") + facet_grid(site_code ~ Year) + theme_mo()
p + theme_mo()
ggsave(filename="~/Dropbox/MRC_with_clades2.pdf", height=14, width=14)
```



```{r, fig.width=10, fig.height=8}

p<-ggplot(mai.mrc,aes(x=Month, fill=clade)) + 
  geom_bar(stat="count", position="fill") + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) + scale_fill_manual(values=rev(mycol60)) + theme(legend.position="bottom") + facet_grid(site_code ~ Year) + theme_tq()
p
ggsave(filename="~/Dropbox/MRC_requires_clades.pdf", height=16, width=22)
```


```{r}
#edit(theme_tq)

theme_mo<-function (base_size = 11, base_family = "") 
{
    blue <- "#2c3e50"
    green <- "#18BC9C"
    white <- "#FFFFFF"
    grey <- "grey80"
    theme_grey(base_size = base_size, base_family = base_family) %+replace% 
        theme(line = element_line(colour = blue, size = 0.5, 
            linetype = 1, lineend = "butt"), rect = element_rect(fill = white, 
            colour = blue, size = 0.5, linetype = 1), text = element_text(family = base_family, 
            face = "plain", colour = blue, size = base_size+5, 
            lineheight = 0.9, hjust = 0.5, vjust = 0.5, angle = 0, 
            margin = margin(), debug = FALSE), axis.line = element_blank(), 
            axis.text = element_text(size = rel(0.8)), axis.ticks = element_line(color = grey, 
                size = rel(1/3)), axis.title = element_text(size = rel(1)), 
            panel.background = element_rect(fill = white, color = NA), 
            panel.border = element_rect(fill = NA, size = rel(1/2), 
                color = blue), panel.grid.major = element_line(color = grey, 
                size = rel(1/3)), panel.grid.minor = element_line(color = grey, 
                size = rel(1/3)), panel.grid.minor.x = element_blank(), 
            panel.spacing = unit(0.02, "cm"), legend.key = element_rect(fill = white, 
                color = NA), legend.position = "right", strip.background = element_rect(fill = blue, 
                color = blue), strip.text = element_text(color = white, 
                size = rel(0.8)), plot.title = element_text(size = rel(1.2), 
                hjust = 0, margin = margin(t = 0, r = 0, b = 4, 
                  l = 0, unit = "pt")), plot.subtitle = element_text(size = rel(0.9), 
                hjust = 0, margin = margin(t = 0, r = 0, b = 3, 
                  l = 0, unit = "pt")), complete = TRUE)
}
```

```{r}

mycol60<-c("#735768",
"#bf989e",
"#a3c6b5",
"#ffd8b9",
"#6a4d2a",
"#5e4a73",
"#d1ffd5",
"#4c5729",
"#cbc5ff",
"#ffb2cb",
"#814036",
"#017e6d",
"#0184b2",
"#71407a",
"#dfffae",
"#20dfff",
"#005aa0",
"#e6b2ff",
"#50fff6",
"#3cb5ff",
"#7b4e00",
"#02d0b6",
"#5f6700",
"#644190",
"#bfa3ff",
"#9c1f4d",
"#006920",
"#4da0ff",
"#01e4b5",
"#ffb65a",
"#a094ff",
"#01b36f",
"#3d7400",
"#933500",
"#ff8160",
"#ffd664",
"#f3ff7f",
"#9dff98",
"#a98900",
"#ff6572",
"#ae0060",
"#b87300",
"#008b2d",
"#53ffb0",
"#a51312",
"#c90041",
"#9bad00",
"#8d1896",
"#a90092",
"#c5bb00",
"#d90085",
"#5739bb",
"#a35ee7",
"#df5c06",
"#53bc30",
"#e7d627",
"#efb601",
"#f663e2",
"#2ae575")
mycols120<-c("#f8279c", "#ffabbd",
"#ff4d84",
"#f10f68",
"#c10050",
"#972944",
"#ff818f",
"#ff5673",
"#904d52",
"#a7042f",
"#de0d3b",
"#ff6767",
"#f03740",
"#ff8975",
"#ff573f",
"#ff9e86",
"#952f1a",
"#863d2a",
"#ff9369",
"#ff7940",
"#9b3800",
"#c25500",
"#ff954f",
"#ffad76",
"#e47b00",
"#964f00",
"#b76500",
"#e9b88f",
"#ff981a",
"#784704",
"#e18800",
"#f4bc79",
"#dc9300",
"#faba5d",
"#b19061",
"#6f4c00",
"#f5b207",
"#caa300",
"#e4c36e",
"#dcc48c",
"#705c00",
"#ab9000",
"#e3c537",
"#8f7b00",
"#575511",
"#cecb43",
"#8b9200",
"#c8cb74",
"#acbb00",
"#576800",
"#87a600",
"#9fa871",
"#b0d25b",
"#618900",
"#3d5b14",
"#3b7b00",
"#9fd57d",
"#69ca3b",
"#329d00",
"#7bdd57",
"#8cd96f",
"#0c6600",
"#1e5f13",
"#5ee061",
"#71dd76",
"#0f6024",
"#007b2d",
"#01c756",
"#73dc89",
"#28603e",
"#00a55f",
"#007e48",
"#01d187",
"#62dbab",
"#01bb94",
"#009475",
"#7bd7ba",
"#007e6c",
"#00c9b0",
"#009993",
"#00c9d6",
"#0298c7",
"#68c5ff",
"#016d9c",
"#43aaff",
"#0077c7",
"#5b7aac",
"#1e5395",
"#9fb6ff",
"#508aff",
"#006de3",
"#224fa1",
"#0159d4",
"#a7abff",
"#949cff",
"#2c48b6",
"#4f4d80",
"#635fe8",
"#9c82ff",
"#6036b6",
"#654583",
"#ad8ec1",
"#da97ff",
"#d37fff",
"#743891",
"#b64cd2",
"#eb8eff",
"#9f27af",
"#ff78f7",
"#862d81",
"#a00099",
"#ffa4ee",
"#c2109f",
"#9a6286",
"#803b64",
"#df0e93",
"#8d2e62",
"#b70076",
"#ff52ad",
"#ff96c5",
"#f8278c")
```

